components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -674,51 +497,128 @@ public void testChgrp(boolean isSudoing, boolean permChgrp,
+         * two of the groups.
+         */
+        final EventContext normalUser = newUserAndGroup(groupPermissions);
+        /* Group where the user is not member */
+        final long anotherGroupId = newUserAndGroup(groupPermissions).groupId;
+        /* Define cases:
+         * When data owner is not member of the target group,
+         * Chgrp action passes only when light admin has Chgrp permission
+         * and is not Sudoing (Sudoing can be thought of as destroyer of the
+         * privileges, because it forces the permissions of the Sudoed-as user
+         * onto the light admin). These permissions should be also sufficient
+         * to move all annotations on the image, which are unique on the image.*/
+        boolean chgrpNonMemberExpectSuccess = !isSudoing && permChgrp;
+        /* define cases where canChgrp on the image is expected to be true */
+        final boolean canChgrpExpectedTrue = permChgrp || isSudoing;
+        /* create a Dataset as the normalUser and import into it */
+        loginUser(normalUser);
+        Dataset dat = mmFactory.simpleDataset();
+        Dataset sentDat = (Dataset) iUpdate.saveAndReturnObject(dat);
+        List<IObject> originalFileAndImage = importImageWithOriginalFile(sentDat);
+        OriginalFile originalFile = (OriginalFile) originalFileAndImage.get(0);
+        Image image = (Image) originalFileAndImage.get(1);
+
+        /* Tag the imported image */
+        TagAnnotation tagAnnotation = new TagAnnotationI();
+        tagAnnotation = (TagAnnotation) iUpdate.saveAndReturnObject(tagAnnotation);
+        final ImageAnnotationLink tagAnnotationLink = annotateImage(image, tagAnnotation);
+        /* add a file attachment with original file to the imported image.*/
+        List<IObject> originalFileAnnotationAndLink = createFileAnnotationWithOriginalFileAndLink(image);
+        final OriginalFile annotOriginalFile = (OriginalFile) originalFileAnnotationAndLink.get(0);
+        final FileAnnotation fileAnnotation = (FileAnnotation) originalFileAnnotationAndLink.get(1);
+        final ImageAnnotationLink fileAnnotationLink = (ImageAnnotationLink) originalFileAnnotationAndLink.get(2);
+
+        /* set up the light admin's permissions for this test */
+        List<String> permissions = new ArrayList<String>();
+        permissions.add(AdminPrivilegeSudo.value);
+        if (permChgrp) permissions.add(AdminPrivilegeChgrp.value);
+        final EventContext lightAdmin;
+        lightAdmin = loginNewAdmin(true, permissions);
+
+        try {
+            sudo(new ExperimenterI(normalUser.userId, false));
+            }catch (SecurityViolation sv) {
+            }
+        /* take care of workflows which do not use sudo */
+        if (!isSudoing) {
+            loginUser(lightAdmin);
+        }
+        /*in order to find the image in whatever group, get context with group
+         * set to -1 (=all groups) */
+        client.getImplicitContext().put("omero.group", Long.toString(-1));
```

Comment:
"-1" suffices instead of the toString -- but now I'm wondering if in your refactoring it's worth creating a,

```java
protected static final ImmutableMap<String, String> ALL_GROUPS_CONTEXT = ImmutableMap.of(Login.OMERO_GROUP, "-1");
```
in the superclass that all the others can use.